### PR TITLE
Implement feature: "sesdev supportconfig DEPLOYMENT_ID NODE_ID"

### PR DESCRIPTION
**Pull request by smithfarm**
Fixes: https://github.com/SUSE/sesdev/issues/75
Signed-off-by: Nathan Cutler <ncutler@suse.com>

---

To Do:

- [ ] assert that it doesn't break `sesdev ssh --recursive`

### CODE REVIEW

**Comment by smithfarm** [comment]
@tserong @toabctl @ricardoasmarques @capri1989 This runs `supportconfig` on a node deployed by sesdev, downloads the resulting tarball to the host, and deletes the tarball on the node (so it doesn't get downloaded again if the command is run multiple times).

Would be very happy if you gave it a spin and a review!

It occurred to me that `sesdev supportconfig` might be a better name than `sesdev report`. What do you think?

**Comment by ricardoasmarques** [comment]
> @tserong @toabctl @ricardoasmarques @capri1989 This runs `supportconfig` on a node deployed by sesdev, downloads the resulting tarball to the host, and deletes the tarball on the node (so it doesn't get downloaded again if the command is run multiple times).
> 
> Would be very happy if you gave it a spin and a review!
> 
> It occurred to me that `sesdev supportconfig` might be a better name than `sesdev report`. What do you think?

`sesdev supportconfig` is indeed more clear/explicit IMO.

File: seslib/__init__.py [hunk 1]
```diff
@@ -1409,8 +1409,23 @@ def _scp_cmd(self, recursive, source, destination):
 
         return _cmd
 
-    def scp(self, recursive, source, destination):
-        tools.run_interactive(self._scp_cmd(recursive, source, destination))
+    def scp(self, source, destination, recurse=False):
+        tools.run_interactive(self._scp_cmd(source, destination, recurse=recurse))
+
+    def report(self, log_handler, name):
+        if self.settings.os.startswith("sle"):
```
Comment (ricardoasmarques) [inline seslib/__init__.py]: nit: Perhaps:
```
if not self.settings.os.startswith("sle"):
    raise SupportconfigOnlyOnSLE()
```
(wihout the `else`) is easier to read

Comment (smithfarm) [reply seslib/__init__.py]: Replaced `pass` with a log message to legitimize this construct. (I find positive conditional expressions easier/faster to grasp.)

File: seslib/exceptions.py [hunk 2]
```diff
@@ -113,3 +113,11 @@ class SettingNotKnown(SesDevException):
     def __init__(self, setting):
         super(SettingNotKnown, self).__init__(
             "Setting '{}' is not known - please open a bug report!".format(setting))
+
+
+class SupportconfigOnlyOnSLE(SesDevException):
+    def __init__(self):
+        super(SupportconfigOnlyOnSLE, self).__init__(
+            "sesdev report depends on supportconfig, which is available only on "
```
Comment (ricardoasmarques) [inline seslib/exceptions.py]: It's no longer `sesdev report`

Comment (smithfarm) [reply seslib/exceptions.py]: fixed

**Pull request merged**